### PR TITLE
Add falcon mamba blogpost 

### PR DESCRIPTION
**Pull request by younesbelkada**
Adds the HF blogpost of FalconMamba model integration, a recent model released by TII 

cc @osanseviero @pcuenca

### CODE REVIEW

**Comment by kashif** [comment]
it's always funny that that the snake in the logos etc. is always a cobra instead of mamba hehe

File: falconmamba.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+An alternative to parallel prefill is to process the prompt token by token, which we will refer to as *sequential prefill*. Akin to sequence parallelism, it can be also done on larger chunks of the prompt instead of individual tokens for better GPU usage. While for transformers sequential prefill makes little sense, it brings back the possibility to process arbitrary long prompts by SSM models. 
+
+With these remarks in mind, we first test the largest sequence length that can fit on a single 24 GB A10 GPU, putting the results on the [figure](#max-length) below. The batch size is fixed at 1 and we are using float32 precision. Even for parallel prefill, FalconMamba can fit larger sequences than a transformer, while in sequential prefill it unlocks its full potential and can process arbitrary long prompt
```
Comment (osanseviero) [inline falconmamba.md:83]:
```suggestion
With these remarks in mind, we first test the largest sequence length that can fit on a single 24 GB A10 GPU, putting the results on the [figure](#max-length) below. The batch size is fixed at 1, and we are using float32 precision. Even for parallel prefill, FalconMamba can fit larger sequences than a transformer, while in sequential prefill, it unlocks its full potential and can process arbitrary long prompt
```

File: falconmamba.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+<a id="max-length"></a>
+![Model Performance](https://huggingface.co/datasets/tiiuae/documentation-images/resolve/main/falcon_mamba/max_len_llalma3-1.png)
+
+Next, we measure the generation throughput in a setting with a prompt of length 1 and up to 130k generated tokens, using batch size 1 and H100 GPU. The results are reported in the [figure](#throughput) below. We observe that our Falcon Mamba is generating all the tokens at constant throughput and without any increase in CUDA peak memory. For the transformer model, the peak memory grows and generation speed slows down as the number of generated tokens grows.
```
Comment (osanseviero) [inline falconmamba.md:88]:
```suggestion
Next, we measure the generation throughput in a setting with a prompt of length 1 and up to 130k generated tokens, using batch size 1 and H100 GPU. The results are reported in the [figure](#throughput) below. We observe that our Falcon Mamba is generating all the tokens at constant throughput and without any increase in CUDA peak memory. For the transformer model, the peak memory grows, and generation speed slows down as the number of generated tokens grows.
```

File: falconmamba.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+<a id="throughput"></a>
+![Model Performance](https://huggingface.co/datasets/tiiuae/documentation-images/resolve/main/falcon_mamba/thoughput-llama3-1.png)
+
+## How to use it within Hugging Face transformers?
+
+The FalconMamba architecture will be available on the next release of Hugging Face transformers library (>4.42.4). Make sure to install the latest version of Hugging Face transformers, or install the library from source to use the model.
```
Comment (osanseviero) [inline falconmamba.md:95]:
```suggestion
The FalconMamba architecture will be available in the next release of the Hugging Face transformers library (>4.45.0). To use the model, make sure to install the latest version of Hugging Face transformers or install the library from the source.
```

File: falconmamba.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+FalconMamba is compatible with most of the APIs Hugging Face offers, which whom you are familiar with, such as `AutoModelForCausalLM` or `pipeline` :
```
Comment (osanseviero) [inline falconmamba.md:97]:
```suggestion
FalconMamba is compatible with most of the APIs Hugging Face offers, which you are familiar with, such as `AutoModelForCausalLM` or `pipeline` : 
```

File: falconmamba.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+model_id = "tiiuae/falcon-mamba-7b" 
+tokenizer = AutoTokenizer.from_pretrained(model_id) 
+
+quantization_config = BitsAndBytesConfig(load_in_4bit=True) 
+model = AutoModelForCausalLM.from_pretrained(model_id, quantization_config=quantization_config) 
+
+inputs = tokenizer("Hello world, today", return_tensors="pt").to(0) 
+output = model.generate(**inputs, max_new_tokens=100, do_sample=True) 
+
+print(tokenizer.decode(Output[0], skip_special_tokens=True))
```
Comment (osanseviero) [inline falconmamba.md:125]:
```suggestion
print(tokenizer.decode(output[0], skip_special_tokens=True)) 
```

File: falconmamba.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+quantization_config = BitsAndBytesConfig(load_in_4bit=True) 
+model = AutoModelForCausalLM.from_pretrained(model_id, quantization_config=quantization_config) 
+
+inputs = tokenizer("Hello world, today", return_tensors="pt").to(0) 
+output = model.generate(**inputs, max_new_tokens=100, do_sample=True) 
+
+print(tokenizer.decode(Output[0], skip_special_tokens=True)) 
+```
+We are also pleased to introduce the instruction-tuned version of FalconMamba, which has been fine-tuned with an additional 5 billion tokens of supervised fine-tuning (SFT) data. This extended training enhances the model's ability to perform instructional tasks with better precision and effectiveness. You can experience the capabilities of the instruct model through our demo, available [here](https://huggingface.co/spaces/tiiuae/falcon-mamba-playground).
```
Comment (osanseviero) [inline falconmamba.md:127]: Maybe add a note on the prompt template

File: falconmamba.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+You can also directly use the 4-bit converted version of both the [base model](https://huggingface.co/tiiuae/falcon-mamba-7b-4bit) and the [instruct model](https://huggingface.co/tiiuae/falcon-mamba-7b-instruct-4bit). Make sure to have access to a GPU that is compatible with `bitsandbytes` library to run the quantized model.
+
+You can also benefit from faster inference using `torch.compile`, simply call `model = torch.compile(model)` once you loaded the model.
```
Comment (osanseviero) [inline falconmamba.md:131]:
```suggestion
You can also benefit from faster inference using `torch.compile`; simply call `model = torch.compile(model)` once you have loaded the model. 
```

File: falconmamba.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+```
+We are also pleased to introduce the instruction-tuned version of FalconMamba, which has been fine-tuned with an additional 5 billion tokens of supervised fine-tuning (SFT) data. This extended training enhances the model's ability to perform instructional tasks with better precision and effectiveness. You can experience the capabilities of the instruct model through our demo, available [here](https://huggingface.co/spaces/tiiuae/falcon-mamba-playground).
+
+You can also directly use the 4-bit converted version of both the [base model](https://huggingface.co/tiiuae/falcon-mamba-7b-4bit) and the [instruct model](https://huggingface.co/tiiuae/falcon-mamba-7b-instruct-4bit). Make sure to have access to a GPU that is compatible with `bitsandbytes` library to run the quantized model.
+
+You can also benefit from faster inference using `torch.compile`, simply call `model = torch.compile(model)` once you loaded the model. 
+
+## Acknowledgments
+
+Authors of this blogpost would like to thank the Hugging Face team for their smooth support and integration within their ecosystem, in particular
```
Comment (osanseviero) [inline falconmamba.md:135]:
```suggestion
The authors of this blog post would like to thank the Hugging Face team for their smooth support and integration within their ecosystem, in particular
```

File: falconmamba.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+
+- [Alina Lozovskaya](https://huggingface.co/alozowski) and [Clementine Fourrier](https://huggingface.co/clefourrier) for helping us evaluating the model on the leaderboard
+- [Arthur Zucker](https://huggingface.co/ArthurZ) for the transformers integration
+- [Vaibhav Srivastav](https://huggingface.co/reach-vb), [hysts](https://huggingface.co/hysts) and [Omar Sanseviero](https://huggingface.co/osanseviero) for their support with questions related to Hub
+
+Authors would also like to thank Tri Dao and Albert Gu for implementing and open-sourcing Mamba architecture to the community.
```
Comment (osanseviero) [inline falconmamba.md:141]:
```suggestion
The authors would also like to thank Tri Dao and Albert Gu for implementing and open-sourcing Mamba architecture to the community.
```

File: falconmamba.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
+---
+title: "Welcome FalconMamba: The first strong attention-free 7B model" 
+thumbnail: /blog/assets/falconmamba/thumbnail.png
+authors:
+- user: JingweiZuo
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply falconmamba.md:5]: Done!

File: falconmamba.md [hunk 11]
```diff
@@ -0,0 +1,141 @@
+---
+title: "Welcome FalconMamba: The first strong attention-free 7B model" 
+thumbnail: /blog/assets/falconmamba/thumbnail.png
+authors:
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply falconmamba.md:4]: Thanks, done !

File: falconmamba.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+- user: JingweiZuo
+- user: yellowvm
+- user: DhiyaEddine
+- user: IChahed
+- user: ybelkada
+- user: Gkunsch
+---
+
+[Falcon Mamba](https://falconllm.tii.ae/) is a new model by [Technology Innovation Institute (TII)](https://www.tii.ae/ai-and-digital-science) in Abu Dhabi released under the TII Falcon License 2.0. The model is open source and available within the Hugging Face ecosystem [here](https://huggingface.co/tiiuae/falcon-mamba-7b) for anyone to use it for their research or application purpose.
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply falconmamba.md]: Just updated all model metadata to `license: - other`

File: falconmamba.md (hunk 6 above)
Comment (younesbelkada) [reply falconmamba.md]: Done also!

File: _blog.yml [hunk 13]
```diff
@@ -4528,3 +4528,15 @@
     - LLM
     - nlp
     - community
+
+- local: falconmamba
+  title: "Welcome FalconMamba: The first strong attention-free 7B model "
+  author: JingweiZuo
```
Comment (osanseviero) [inline _blog.yml:4534]:
```suggestion
  guest: true
  author: JingweiZuo
```

**Review by osanseviero — APPROVED** [review]
Cool :fire:

**Pull request merged**